### PR TITLE
Javac plugin occurrences

### PR DESCRIPTION
**Pull request by alexander-myltsev**
Add occurrences generation to Javac plugin

Possible improvements for the next tasks:
- add occurrences for type usages 
- add occurrences for classes nested within methods 
- add occurrences for anonymous objects

### CODE REVIEW

File: semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/SemanticdbGen.scala [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
 import scala.meta.internal.{semanticdb => s}
-import scala.meta.internal.semanticdb.Scala._
 import scala.meta.internal.semanticdb.javac.semantics._
 import scala.meta.internal.io.PathIO
 
-class SemanticdbGen(relSourcePath: Path, toplevels: Seq[TypeElement]) {
+class SemanticdbGen(
+    relSourcePath: Path,
+    toplevels: Seq[TypeElement],
+    val trees: Trees,
```
[the remark this answers is not in the document]
Comment (alexander-myltsev) [reply semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/SemanticdbGen.scala]: Nowhere. Good catch!

File: semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Semantics.scala [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+              (methodPlace: @switch) match {
+                case -1 => ???
+                case 0 => "()"
+                case x => s"(+$x)"
+              }
+            }
+            Symbols.Global(owner.sym, d.Method(symbolName, disambig))
+
+          case Some(owner: jp.ast.expr.ObjectCreationExpr) =>
+            Symbols.Global(s"new_${owner.getType.toString}", d.Type(symbolName))
```
[the remark this answers is not in the document]
Comment (alexander-myltsev) [reply semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Semantics.scala]: I introduced that to correctly handle methods inside within anonymous objects:

```scala
public <Anon> ArrayList<Anon> foo() {
        return new ArrayList<Anon>() {
            @Override
            public boolean remove(Object o) {
                return true;
            }
        };
    }
```

I replace it with the empty string for the current commit.

File: semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Elements.scala [hunk 3]
```diff
@@ -206,6 +207,30 @@ trait Elements {
       }
     }
 
+    def range: Option[s.Range] = {
```
[the remark this answers is not in the document]
Comment (alexander-myltsev) [reply semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Elements.scala]: The `range` is defined within `implicit class Element` context. I don't see much sense in renaming it. Though, `nameRange` is equally fine. You could insist :) Do you still prefer to see it renamed?

[the remark this answers is not in the document]
Comment (xeno-by) [reply semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Elements.scala]: No strong preference.

**Comment by olafurpg** [comment]
Thank you for this contribution @alexander-myltsev ! As discussed offline, this PR is superseded by the new repository https://github.com/alexander-myltsev/scalameta-java/ I am excited to have the ability to generate SemanticDB occurrences for Java!

**Pull request closed without merging**